src/lustre/lustreLexer.mll

```diff
@@ -334,7 +335,10 @@ let keyword_table = mk_hashtbl [
    C syntax: alphanumeric characters including the underscore, starting 
    with a letter or the underscore *)
 let id = ['a'-'z' 'A'-'Z' '_'] ['a'-'z' 'A'-'Z' '_' '0'-'9']*
-         
+
+let qual_id = id (':' ':' id)*
+(* a qualified id is an id or id::qual_id *)
+                
```

Comment:
A qualified identifier shouldn’t be a single token, but a grammar rule. This forces the parser to deconstruct the qualified identifier unnecessarily.